bootstrap/bootstrap.go

```diff
@@ -890,38 +893,98 @@ func (b *Bootstrap) defaultCommandPhase() error {
 		return fmt.Errorf("This agent is only allowed to run scripts within your repository. To allow this, re-run this agent without the `--no-command-eval` option, or specify a script within your repository to run instead (such as scripts/test.sh).")
 	}
 
-	shellArgs, err := shell.Parse(b.Shell)
+	var cmdToExec string
+
+	// The shell gets parsed based on the operating system
+	shell, err := shellwords.Split(b.Shell)
 	if err != nil {
-		return err
+		return fmt.Errorf("Failed to split shell (%q) into tokens: %v", b.Shell, err)
 	}
 
-	var cmd []string
+	if len(shell) == 0 {
```

Comment:
What sort of shell values can be passed to cause it to get a `len` of 0?

Reply:
An empty string :)